fix: corrected assertions of the scan verb test

purnimavenkatasubbu opened the pull request:
**- What we did**
* Changed the incorrect assertions
* Added test case for keys with mixed case - discovered #1216 
* Fixed #1216

**- How we did it**
* Removed the update: public command in the scan result assertion
* Ensure HiveKeystore public methods start by converting their 'key' parameter to lower case. This ensures that the _metaDataCache's keys are in lower-case, same as the keyStore itself.

**- How to verify it**
* all tests should pass

purnimavenkatasubbu committed:
updated assertions correctly

purnimavenkatasubbu committed:
Empty-Commit

gkc commented on tests/at_functional_test/test/scan_verb_test.dart:
```diff
@@ -120,18 +120,31 @@ void main() {
     ///UPDATE VERB
     await socket_writer(
-        socketFirstAtsign!, 'update:public:verifyingTTL$firstAtsign Working?');
+        socketFirstAtsign!, 'update:ttl:3000:ttlkey.me$firstAtsign 1245');
```
Can you modify this test so that the entity is 'ttlKey.me' (i.e. capitalize the k) and modify the expects accordingly? Would be good to be asserting that in the scan response, the entity is in lowercase

purnimavenkatasubbu replied:
sure

purnimavenkatasubbu replied:
Updated the test with a capital K. But the scan response doesn't show up the key. Raised a bug for the same.
[bug](https://github.com/atsign-foundation/at_server/issues/1216)

gkc replied:
You haven't changed the expects to match :-)

gkc replied:
Oh I see, you didn't commit the changes

purnimavenkatasubbu replied:
Yes @gkc . I haven't pushed my changes because the test will fail in actions

gkc committed:
fix: fixes #1216
Ensure HiveKeystore public methods start by converting their 'key' parameter to lower case. This ensures that the _metaDataCache's keys are in lower-case, same as the keyStore itself.

gkc commented:
@purnimavenkatasubbu I'm pushing the fix for #1216 to this PR. I've updated the scan_verb_test also. Once the checks have passed, I will publish new at_persistence_secondary_server, remove the dependency override from at_secondary_server's pubspec, push the branch again, wait for checks to pass and then merge to trunk.

purnimavenkatasubbu commented:
> @purnimavenkatasubbu I'm pushing the fix for #1216 to this PR. I've updated the scan_verb_test also. Once the checks have passed, I will publish new at_persistence_secondary_server, remove the dependency override from at_secondary_server's pubspec, push the branch again, wait for checks to pass and then merge to trunk.

Thanks

gkc committed:
build: Update at_persistence_secondary_server CHANGELOG. Remove dependency override from at_secondary_server's pubspec.yaml
